pkg/logs/sender/message_buffer.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package sender
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// MessageBuffer accumulates lines in buffer escaping all '\n'
+// and accumulates the total number of bytes of all lines in line representation (line + '\n') in rawDataLen
+type MessageBuffer struct {
```

Comment:
This class has two responsibilities, buffering messages and building the payload and I feel like we should rename it as well as define a clear API.
For example:
```
type BatchPayloadBuilder interface {
  func TryAddMessage(msg message.Message) bool
  func Build() []byte
  func Clear()
  func GetMessages() [] message.Message
  func IsEmpty() bool
  func IsFull() bool
}
```